### PR TITLE
Added ES5 polyfill for Array find method.

### PR DESCRIPTION
**Pull request by hubertkozik**
Fixing ES5 code generation.

### CODE REVIEW

**Review by TorsteinHonsi — APPROVED** [review]
Thanks, looks good!

File: ts/masters-es5/highcharts.src.ts [hunk 1]
```diff
@@ -24,15 +24,42 @@ if (MSPointer.isRequired()) {
...
+if (!Array.prototype.find) {
+    // eslint-disable-next-line no-extend-native
+    Array.prototype.find = function <T> (
+        predicate: (search: T, index: number, array: any[]) => boolean
+    ): T | undefined {
+        if (this === null) {
+            throw new TypeError(
+                'Array.prototype.find called on null or undefined'
+            );
+        }
```
Comment (bre1470) [inline ts/masters-es5/highcharts.src.ts:44]: We can remove it as we assign it to the prototype.

File: ts/masters-es5/highcharts.src.ts [hunk 2]
```diff
@@ -24,15 +24,42 @@ if (MSPointer.isRequired()) {
...
+        if (typeof predicate !== 'function') {
+            throw new TypeError('predicate must be a function');
+        }
```
Comment (bre1470) [inline ts/masters-es5/highcharts.src.ts:47]: This is also something that is not necessary and could fail hard without much difference to functionality.

File: ts/masters-es5/highcharts.src.ts [hunk 3]
```diff
@@ -24,15 +24,42 @@ if (MSPointer.isRequired()) {
...
+        const list = Object(this),
```
Comment (bre1470) [inline ts/masters-es5/highcharts.src.ts:48]: Why is this needed?

File: ts/masters-es5/highcharts.src.ts [hunk 4]
```diff
@@ -24,15 +24,42 @@ if (MSPointer.isRequired()) {
...
+            length = list.length >>> 0,
```
Comment (bre1470) [inline ts/masters-es5/highcharts.src.ts:49]: Trimming the length is a bit strange as length will never be too big or negative.

File: ts/masters-es5/highcharts.src.ts [hunk 5]
```diff
@@ -24,15 +24,42 @@ if (MSPointer.isRequired()) {
...
+            thisArg = arguments[1];
```
Comment (bre1470) [inline ts/masters-es5/highcharts.src.ts:50]: = scope parameter. This should go into the `find` parameters as the optional second, see [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#thisarg).

File: ts/masters-es5/tsconfig.json [hunk 6]
```diff
@@ -13,7 +13,8 @@
             "ES5",
             "ES2017.Object",
             "ES2015.Promise",
-            "ES2016.Array.Include"
+            "ES2016.Array.Include",
+            "ES2015"
```
Comment (bre1470) [inline ts/masters-es5/tsconfig.json:17]: Just core please: `ES2015.Core`

File: ts/masters-es5/highcharts.src.ts (hunk 3 above)
Comment (hubertkozik) [reply ts/masters-es5/highcharts.src.ts:48]: To be honest, I've copied the polyfill from somewhere 😁 I thought, that for polyfill there could be any possibility, that casting to Object is needed.

**Comment by hubertkozik** [comment]
I've refactored it with the requested changes. Do you think it's good enough @bre1470?
